### PR TITLE
Use the correct type for failure details

### PR DESCRIPTION
**Pull request by ylil93**
- make sure status is a dict for pairwise result when handling an internal error

### CODE REVIEW

File: compatibility_lib/compatibility_lib/compatibility_store.py [hunk 1]
```diff
@@ -232,6 +232,9 @@ def __init__(self,
         # are None.
         if mysql_host is None and mysql_unix_socket is None:
             mysql_host = '127.0.0.1'
+            assert mysql_user is not None
```
Comment (brianquinlan) [inline compatibility_lib/compatibility_lib/compatibility_store.py]: Could you put this unrelated change in another PR?

Comment (ylil93) [reply compatibility_lib/compatibility_lib/compatibility_store.py]: done
